### PR TITLE
Refs #29522 -- Improved test coverage of deserializers.

### PR DESCRIPTION
**Pull request by sarahboyce**
Cherry picked from #18335

### CODE REVIEW

File: tests/fixtures_regress/fixtures/sequence_empty_lines_jsonl.jsonl [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{"pk": "1", "model": "fixtures_regress.animal", "fields": {"name": "Eagle", "latin_name": "Aquila", "count": 3, "weight": 1.2}}
+
+
```
[the remark this answers is not in the document]
Comment (sarahboyce) [reply tests/fixtures_regress/fixtures/sequence_empty_lines_jsonl.jsonl:3]: Yes the test covers handling of empty lines :+1:

[the remark this answers is not in the document]
Comment (sarahboyce) [reply tests/fixtures_regress/fixtures/sequence_empty_lines_jsonl.jsonl:3]: Should I make the starting lines empty so it looks less like a formatting error?

[the remark this answers is not in the document]
Comment (pauloxnet) [reply tests/fixtures_regress/fixtures/sequence_empty_lines_jsonl.jsonl:3]: Maybe

File: tests/fixtures_regress/tests.py [hunk 2]
```diff
@@ -96,12 +96,18 @@ def test_loaddata_not_found_fields_not_ignore(self):
...
+            "sequence_extra_jsonl",
+            "sequence_extra_yaml",
+        ):
+            with self.subTest(fixture_file=fixture_file):
+                with self.assertRaises(DeserializationError):
+                    management.call_command(
+                        "loaddata",
+                        fixture_file,
+                        verbosity=0,
+                    )
```
Comment (felixxm) [inline tests/fixtures_regress/tests.py:110]: Nitpick
```suggestion
            with (
                self.subTest(fixture_file=fixture_file),
                self.assertRaises(DeserializationError),
            ):
                management.call_command(
                    "loaddata",
                    fixture_file,
                    verbosity=0,
                )
```

**Pull request merged**